src/ar_storage.erl

```diff
@@ -29,16 +29,17 @@ start() ->
 
 %% @doc Ensure that all of the relevant storage directories exist.
 ensure_directories() ->
-	filelib:ensure_dir(?TX_DIR),
-	filelib:ensure_dir(?BLOCK_DIR),
-	filelib:ensure_dir(?BLOCK_ENC_DIR),
-	filelib:ensure_dir(?WALLET_LIST_DIR),
-	filelib:ensure_dir(?HASH_LIST_DIR),
+	DataDir = ar_meta_db:get(data_dir) ++ "/",
+	filelib:ensure_dir(DataDir ++ ?TX_DIR ++ "/"),
+	filelib:ensure_dir(DataDir ++ ?BLOCK_DIR ++ "/"),
+	filelib:ensure_dir(DataDir ++ ?BLOCK_ENC_DIR ++ "/"),
+	filelib:ensure_dir(DataDir ++ ?WALLET_LIST_DIR ++ "/"),
```

Comment:
The slash in the end should not be needed. We have it in many places.

Reply:
It is in fact needed here, to make `filelib:ensure_dir` create a directory if one does not exist.

Reply:
Ohhh I made false assumptions ✅